incubator/virtualcluster/pkg/syncer/resources/crd/checker.go

```diff
@@ -0,0 +1,175 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+ Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+     http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package crd
+
+import (
+	"context"
+	"fmt"
+	"k8s.io/apimachinery/pkg/util/sets"
+	apiextensionsclientset "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/klog"
+	"sigs.k8s.io/multi-tenancy/incubator/virtualcluster/pkg/syncer/constants"
+	"sigs.k8s.io/multi-tenancy/incubator/virtualcluster/pkg/syncer/conversion"
+	"sigs.k8s.io/multi-tenancy/incubator/virtualcluster/pkg/syncer/metrics"
+	"sigs.k8s.io/multi-tenancy/incubator/virtualcluster/pkg/syncer/patrol/differ"
+	"sync/atomic"
+)
+
+var numMissMatchedCRD uint64
+
+func (c *controller) StartPatrol(stopCh <-chan struct{}) error {
+	defer utilruntime.HandleCrash()
+	if !cache.WaitForCacheSync(stopCh, c.crdSynced, c.vcSynced) {
+		return fmt.Errorf("failed to wait for caches to sync before starting networkpolicy checker")
+	}
+	c.crdPatroller.Start(stopCh)
+	return nil
+}
+
+// PatrollerDo checks to see if annotated CRD is in super master informer cache and then synced to tenant cluster
+func (c *controller) PatrollerDo() {
+	clusterNames := c.multiClusterCrdController.GetClusterNames()
+	if len(clusterNames) == 0 {
+		klog.Infof("tenant masters has no clusters, give up period checker")
+		return
+	}
+
+	pList := &v1beta1.CustomResourceDefinitionList{}
+	err := c.superClient.List(context.Background(), pList)
+	if err != nil {
+		klog.Errorf("error listing crd from super master informer cache: %v", err)
+		return
+	}
+
+	pSet := differ.NewDiffSet()
+	for i := range pList.Items {
+		if publicCRD(&pList.Items[i]) {
+			pSet.Insert(differ.ClusterObject{Object: &pList.Items[i], Key: CRDClusterObjectKey(&pList.Items[i], "")})
+		}
+	}
+
+	blockedClusterSet := sets.NewString()
```

Comment:
Can you please revise the use of blockedClusterSet following the latest  PR https://github.com/kubernetes-sigs/multi-tenancy/pull/1418.   (I don't think the bug in 1418 applies for CRD resource, but it is preferred to make code consistent).

Reply:
Sure, I saw the latest change...